### PR TITLE
DEIMOS tuneup

### PR DESCRIPTION
**Pull request by debora-pe**
- DEIMOS tuneup of a few parameters for a better reduction. Specifically changed order of the fits for the tilt and decreased `sigdetect` for wavelength calibration.
- Fix a problem with slitmask matching in edgetrace.py
- Remove sobj if both OPT_COUNTS and BOX_COUNTS are None. Also, give a warning when OPT_COUNTS is None

### CODE REVIEW

File: pypeit/edgetrace.py [hunk 1]
```diff
@@ -4267,7 +4267,7 @@ def maskdesign_matching(self, debug=False):
         # Find if there are missing traces.
         # Need exactly one occurrence of each index in "need"
         buffer = self.par['det_buffer']+1
-        need = ((top_edge_pred > buffer) & (bot_edge_pred < (self.traceimg.shape[1] - 1 - buffer))) & \
+        need = (top_edge_pred > 3*buffer) & (bot_edge_pred < (self.traceimg.shape[1] - 1 - 3*buffer)) & \
```
Comment (profxj) [inline pypeit/edgetrace.py]: Should this be a parameter?

File: pypeit/extraction.py [hunk 2]
```diff
@@ -283,6 +283,20 @@ def extract(self, global_sky, sobjs_obj):
                                           model_noise=(not self.bkg_redux),
                                           show_profile=self.extract_show,
                                           show=self.extract_show)
+        embed()
```
Comment (profxj) [inline pypeit/extraction.py]: remove the `embed()`

**Review by profxj — APPROVED** [review]
Looks good.  Please run tests on DEIMOS in the DevSuite
and confirm things look ok.

**Review by kbwestfall — APPROVED** [review]
Looks good!

**Comment by debora-pe** [comment]
Run the Dev Suite and all passed except the usual Gemini/GNIRS coadd1d wrong OBJID:

```
--- PYPEIT DEVELOPMENT SUITE FAILED 1/142 TESTS (Masters ignored) ---
Failed tests:
    gemini_gnirs/32_SB_SXD pypeit_coadd_1dspec
Skipped tests:
    gemini_gnirs/32_SB_SXD pypeit_coadd_2dspec
    gemini_gnirs/32_SB_SXD pypeit_tellfit
Testing Started at 2022-05-18T15:56:53.103189
Testing Completed at 2022-05-19T01:31:07.215610
Total Time: 9:34:14.112421
```

```
gemini_gnirs/32_SB_SXD pypeit_coadd_1dspec Result: --- FAILED

Logfile:    /Users/dpelliccia/PypeIt/PypeIt-development-suite/REDUX_OUT/gemini_gnirs/32_SB_SXD/gemini_gnirs_32_sb_sxd.test_1dcoadd.log
Process Id: 23322
Start time: Wed May 18 22:40:50 2022
End time:   Wed May 18 22:40:54 2022
Duration:   0:00:03.419799
Command:    pypeit_coadd_1dspec /Users/dpelliccia/Pypeit/PypeIt-development-suite/coadd1d_files/gemini_gnirs_32_sb_sxd.coadd1d

Error Messages:

End of Log:
  File "/Users/dpelliccia/PypeIt/PypeIt/pypeit/pypmsgs.py", line 185, in error
    raise PypeItError(msg)
pypeit.pypmsgs.PypeItError: No matching objects for OBJ0211-DET01.  Odds are you input the wrong OBJID
```

**Review by jhennawi — APPROVED** [review]
Thanks!

**Pull request merged**